kyo-core/shared/src/test/scala/kyoTest/abortsTest.scala

```diff
@@ -255,6 +263,32 @@ class abortsTest extends KyoTest:
                             Left(ex1)
                     )
                 }
+                "distinct" in {
+                    class Distinct1 extends Throwable derives CanEqual
+                    val d1 = new Distinct1
+                    class Distinct2 extends Throwable derives CanEqual
+                    val d2 = new Distinct2
+
+                    val distinct: Boolean < Aborts[Distinct1 | Distinct2] =
+                        for
+                            _ <- Aborts.catching[Distinct1](throw d1)
+                            _ <- Aborts.catching[Distinct2](throw d2)
+                        yield true
+
+                    val a = Aborts.run[Distinct1](distinct)
+                    val b = Aborts.run[Distinct2](a).pure
+                    assert(b == Right(Left(d1)))
+
+                    val c = Aborts.run(distinct).pure
+                    assert(c == Left(d1))
+                }
+                "ClassTag inference" in pendingUntilFixed {
+                    assertCompiles("""
+                        val r = Aborts.run(Aborts.catching(throw new RuntimeException)).pure
```

Comment:
What happens here? Does the `ClassTag` get inferred to `Nothing`/`Object`?

Reply:
```
[error] -- [E172] Type Error:
[error] 286 |                    val r = Aborts.run(Aborts.catching(throw new RuntimeException)).pure
[error]     |                                                                                  ^
[error]     |           No ClassTag available for V
[error]     |
[error]     |           where:    V is a type variable with constraint <: Throwable
[error] one error found
[error] (Test / compileIncremental) Compilation failed
```